firebase-firestore/src/main/java/com/google/firebase/firestore/grpc/DefaultGrpcMetadata.java

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.grpc;
+
+import androidx.annotation.NonNull;
+import com.google.firebase.heartbeatinfo.HeartBeatInfo;
+import com.google.firebase.platforminfo.UserAgentPublisher;
+import io.grpc.Metadata;
+
+/**
+ * Class provides an implementation of the GrpcMetadata interface.
+ *
+ * <p>This updates the metadata with platformInfo string and the heartBeatInfo code.
+ */
+public class DefaultGrpcMetadata implements GrpcMetadata {
```

Comment:
`DefaultGrpcMetadata` is not very descriptive. Since this is related to `x-firebase-client*` headers could we use `FirebaseClient` instead of `Default`?

Other alternatives:
  * `HeartBeat`-  -- though this isn't quite right because this handles both the user agent and heartbeat headers.